src/storage/src/hummock/compactor/mod.rs

```diff
@@ -557,58 +591,73 @@ impl Compactor {
-                        },
+                        event = response_event_stream.next() => {
+                            event
+                        }
+
                         _ = &mut shutdown_rx => {
                             tracing::info!("Compactor is shutting down");
                             return
                         }
                     };
-                    match message {
-                        Some(Ok(SubscribeCompactTasksResponse { task })) => {
-                            let task = match task {
-                                Some(task) => task,
+
+                    match event {
+                        Some(Ok(SubscribeCompactionEventResponse { event })) => {
+                            let event = match event {
+                                Some(event) => event,
                                 None => continue 'consume_stream,
                             };
 
                             let shutdown = shutdown_map.clone();
                             let context = compactor_context.clone();
                             let meta_client = hummock_meta_client.clone();
-
                             executor.spawn(async move {
-                                match task {
-                                    Task::CompactTask(compact_task) => {
+                                let running_task_count = running_task_count.clone();
+                                match event {
+                                    ResponseEvent::CompactTask(compact_task)  => {
+                                        running_task_count.fetch_add(1, Ordering::SeqCst);
                                         let (tx, rx) = tokio::sync::oneshot::channel();
                                         let task_id = compact_task.task_id;
                                         shutdown.lock().unwrap().insert(task_id, tx);
-                                        Compactor::compact(context, compact_task, rx).await;
+                                        let (compact_task, table_stats) = Compactor::compact(context, compact_task, rx).await;
                                         shutdown.lock().unwrap().remove(&task_id);
+                                        running_task_count.fetch_sub(1, Ordering::SeqCst);
+
+                                        if let Err(e) = request_sender.send(SubscribeCompactionEventRequest {
+                                            event: Some(RequestEvent::ReportTask(
+                                                ReportTask {
+                                                    compact_task: Some(compact_task),
+                                                    table_stats_change:to_prost_table_stats_map(table_stats),
+                                                }
+                                            )),
+                                        }) {
+                                            tracing::warn!("Failed to report task {task_id:?} . {e:?}");
```

Comment:
Does send failure indicate stream is broken and need a restart?

Reply:
Where is the rebuild logic when ReportTask failed?